src/main/java/fi/helsinki/cs/tmc/cli/backend/Settings.java

```diff
@@ -116,8 +126,58 @@ public Path getConfigRoot() {
     }
 
     @Override
-    public void setCourse(Course course) {}
+    public String hostProgramName() {
+        // which command line is used
+        return null;
+    }
+
+    @Override
+    public String hostProgramVersion() {
+        // which command line is used
+        return null;
+    }
 
     @Override
-    public void setConfigRoot(Path path) {}
+    public boolean getSendDiagnostics() {
+        return false;
```

Comment:
This should be an user-controlled setting. If it's always false, we won't ever get crash reports.

Reply:
There is no setter for this method, so how is the behaviour determined?